base/lib/stellar/operation.rb

```diff
@@ -335,13 +335,19 @@ def allow_trust(attributes = {})
         end
 
         raise ArgumentError, "Bad :trustor" unless trustor.is_a?(Stellar::KeyPair)
-        raise ArgumentError, "Bad :authorize" unless authorize == !!authorize # check boolean
+
+        op.authorize = if authorize == !!authorize # check boolean
+          authorize ? TrustLineFlags.authorized_flag : 0
+        else
+          raise ArgumentError, "Bad :authorize" if authorize > TrustLineFlags.authorized_to_maintain_liabilities_flag.value
+          authorize
+        end
```

Comment:
I think there's a missing `.value` for authorized flag, but maybe we can go further and provide a bit cleaner interface here?

```suggestion
        op.authorize = case authorize
          when :none, false then 0
          when :full, true then TrustLineFlags.authorized_flag.value
          when :maintain_liabilities then TrustLineFlags.authorized_to_maintain_liabilities_flag.value          
          else
             raise ArgumentError, "Bad :authorize, supported values: :full, :maintain_liabilities, :none"
        end
```

Reply:
Good catch! Added your code with some specs